funcx_sdk/funcx/serialize/concretes.py

```diff
@@ -124,16 +128,87 @@ class DillCode(SerializeBase):
     def __init__(self):
         super().__init__()
 
-    def serialize(self, data):
+    def serialize(self, data) -> str:
         x = codecs.encode(dill.dumps(data), "base64").decode()
         return self.identifier + x
 
-    def deserialize(self, payload):
+    def deserialize(self, payload: str):
         chomped = self.chomp(payload)
         function = dill.loads(codecs.decode(chomped.encode(), "base64"))
         return function
 
 
+class CombinedCode(SerializeBase):
+    """This method uses multiple methods to serialize a function
+
+    # These should be a superset of the methods
+    Code from interpreter/main        : Yes
+    Code from notebooks               : Yes
+    Works with mismatching py versions: Yes
+    Decorated fns                     : Yes
+    """
+
+    identifier = "10\n"
+
+    def get_multiple_payloads(self, payload: str) -> Tuple:
+        parts = self.chomp(payload).split(COMBINED_SEPARATOR)
+        assert len(parts) == 2 * len(DO_SERIALIZE_METHODS)
+        payloads = []
+        for index in range(len(parts) // 2):
+            payloads.append([parts[index], parts[index + 1]])
+
+        # ie. [['04\n', 'gACQ...'], ['01\n', 'sAAbk2...'], ...]
+        return payloads
+
+    def serialize(self, data) -> str:
+        chunks = []
+        for id in DO_SERIALIZE_METHODS:
+            method = METHODS_MAP_CODE[id]()
+            serialized = method.serialize(data)
+            id_length = len(id)
+            chunks.append(serialized[:id_length])
+            chunks.append(serialized[id_length:])
+        return self.identifier + COMBINED_SEPARATOR.join(chunks)
+
+    def deserialize(self, payload: str, variation: int = 1):
+        """
+        Return the other variants, if encoding contains them,
+        if requested.  varation is 1-indexed (ie. 2 means 2nd method)
+        """
+        count = 0
+        for serial_id, encoded_func in self.get_multiple_payloads(payload):
+            count += 1
+            if count == variation:
+                deserialize_method = METHODS_MAP_CODE[serial_id]()
+                try:
+                    return deserialize_method.deserialize(serial_id + encoded_func)
+                except Exception:
+                    logger.info(f"Failed to deserialize using method #{count}")
```

Comment:
Given the `if count == variation`, if this Exception branch occurs, there's no sense in iterating the rest of the loop, yeah?

But given the text of the `raise` statement below, the implication is that we have actually tried to deserialize multiple times.  Am I missing a step in how this loop functions?